### PR TITLE
add ability to change log level in installer

### LINKED ISSUE #29: Add LogLevel support for library logging
Add the ability to set the log level in the Gandalf installer.

### PR DESCRIPTION
**Pull request by jsibbold**
Closes #29

### CODE REVIEW

**Comment by stkent** [comment]
Hmm, I'm not sure if this is what was intended for issue #33 . I believe the plan was something like:

- set log level when installing;
- have all logging calls routed through a centralized class;
- have that centralized class print log statements only if the specified log level is more verbose than the requested log level.

e.g. the logging class methods would look something like this pseudo-code:

```
public static void e(String message) {
    if (this.specifiedLogLevel <is equal to or more verbose than> LogLevel.ERROR) {
        Log.e(TAG, message);
    } else {
        // no-op
    }
}
```

**Comment by jsibbold** [comment]
I don't really have objections to that approach, but right now the enums seem arbitrarily ordered for that to work, I would think we would want NONE, DEBUG, ERROR in that order if we wanted it to work the way you're thinking.

File: gandalf/src/main/java/io/github/btkelly/gandalf/utils/LoggerUtil.java [hunk 1]
```diff
@@ -23,26 +23,35 @@
 public final class LoggerUtil {
 
     private static final String LOGGER_TAG = "Gandalf";
+    private static LogLevel level = LogLevel.NONE;
 
     public enum LogLevel {
```
Comment (jsibbold) [inline gandalf/src/main/java/io/github/btkelly/gandalf/utils/LoggerUtil.java]: are you guys open to changing these to @IntDef?

Comment (stkent) [inline gandalf/src/main/java/io/github/btkelly/gandalf/utils/LoggerUtil.java]: Yep, ok with me.

Comment (btkelly) [inline gandalf/src/main/java/io/github/btkelly/gandalf/utils/LoggerUtil.java]: If we do an int def will the parameter still auto complete to the types when using the Gandalf installer? If setting the log level just becomes an integer I would like to keep it as an enum.

Comment (jsibbold) [inline gandalf/src/main/java/io/github/btkelly/gandalf/utils/LoggerUtil.java]: well we apply the @LogLevel annotation to the method parameter for type safety in this case

Comment (jsibbold) [inline gandalf/src/main/java/io/github/btkelly/gandalf/utils/LoggerUtil.java]: just checked the autocomplete, indeed it does, all hail Android Studio

File: gandalf/src/main/java/io/github/btkelly/gandalf/Gandalf.java [hunk 2]
```diff
@@ -162,6 +165,11 @@ public Installer setPackageName(String packageName) {
             return this;
         }
 
+        public Installer setLogLevel(@NonNull @LogLevel final int logLevel) {
```
Comment (stkent) [inline gandalf/src/main/java/io/github/btkelly/gandalf/Gandalf.java]: `@NonNull` needed here?

File: gandalf/src/main/java/io/github/btkelly/gandalf/utils/LoggerUtil.java [hunk 3]
```diff
@@ -24,25 +28,37 @@
...
+     */
+    public static void setLogLevel(@LogLevel final int level) {
+        LoggerUtil.logLevel = level;
+    }
+
     public static void logD(String message) {
-        //TODO add log level config to gandalf class
-        Log.d(LOGGER_TAG, message);
+        if (logLevel >= DEBUG) {
+            Log.d(LOGGER_TAG, message);
```
Comment (stkent) [inline gandalf/src/main/java/io/github/btkelly/gandalf/utils/LoggerUtil.java]: This will only log debug messages; shouldn't it log both error and debug?

Comment (stkent) [inline gandalf/src/main/java/io/github/btkelly/gandalf/utils/LoggerUtil.java]: Oh wait...

Comment (stkent) [inline gandalf/src/main/java/io/github/btkelly/gandalf/utils/LoggerUtil.java]: I see now.

File: gandalf/src/main/java/io/github/btkelly/gandalf/Gandalf.java (hunk 2 above)
Comment (jsibbold) [inline gandalf/src/main/java/io/github/btkelly/gandalf/Gandalf.java]: doh

Comment (jsibbold) [inline gandalf/src/main/java/io/github/btkelly/gandalf/Gandalf.java]: fixed

**Comment by stkent** [comment]
This looks good, but can't be merged until #34 is resolved.

**Comment by stkent** [comment]
Approved

**Comment by stkent** [comment]
LGTM on success!

**Comment by btkelly** [comment]
Just needs a master merge

**Comment by stkent** [comment]
@jsibbold needs merge

**Comment by btkelly** [comment]
Approved

**Pull request merged**